src/toil/wdl/wdltoil.py

```diff
@@ -5226,65 +5448,107 @@ def run(self, file_store: AbstractFileStore) -> Promised[WDLBindings]:
+
+    This class is only used when runImportsOnWorkers is enabled.
+    """
+
     def __init__(
         self,
-        target: WDL.Tree.Workflow | WDL.Tree.Task,
+        target: Union[WDL.Tree.Workflow, WDL.Tree.Task],
         inputs: WDLBindings,
         wdl_options: WDLContext,
-        path: list[str] | None = None,
-        skip_remote: bool = False,
-        disk_size: ParseableIndivisibleResource | None = None,
+        inputs_search_path: List[str],
+        import_remote_files: bool,
+        import_workers_threshold: ParseableIndivisibleResource,
         **kwargs: Any,
     ):
         """
         Job to take the inputs from the WDL workflow and import them on a worker instead of a leader. Assumes all local and cloud files are accessible.
 
         This class is only used when runImportsOnWorkers is enabled.
         """
-        super().__init__(wdl_options=wdl_options, local=False, disk=disk_size, **kwargs)
-        self._target = target
+        super().__init__(local=True, wdl_options=wdl_options, **kwargs)
         self._inputs = inputs
-        self._path = path
-        self._skip_remote = skip_remote
+        self._target = target
+        self._inputs_search_path = inputs_search_path
+        self._import_remote_files = import_remote_files
+        self._import_workers_threshold = import_workers_threshold
 
     def run(self, file_store: AbstractFileStore) -> Promised[WDLBindings]:
-        """
-        Import the workflow inputs and then create and run the workflow.
-        :return: Promise of workflow outputs
-        """
-        imported_inputs = convert_remote_files(
-            self._inputs,
+        filenames = extract_workflow_inputs(self._inputs)
+        file_to_data = get_file_sizes(
+            filenames,
             file_store.jobStore,
-            self._target.name,
-            self._path,
-            self._skip_remote,
-            self._wdl_options.get("execution_dir")
+            self._inputs_search_path,
+            import_remote_files=self._import_remote_files,
+        )
+        imports_job = ImportsJob(file_to_data, self._import_workers_threshold)
+        self.addChild(imports_job)
+        install_imports_job = WDLInstallImportsJob(
+            self._target.name, self._inputs, imports_job.rv()
         )
-        root_job = WDLStartJob(
-            self._target, imported_inputs, wdl_options=self._wdl_options
+        self.addChild(install_imports_job)
+        imports_job.addFollowOn(install_imports_job)
+
+        start_job = WDLStartJob(
+            self._target, install_imports_job.rv(), wdl_options=self._wdl_options
         )
-        self.addChild(root_job)
-        return root_job.rv()
+        self.addChild(start_job)
+        install_imports_job.addFollowOn(start_job)
+
+        return start_job.rv()
 
 
 def make_root_job(
     target: WDL.Tree.Workflow | WDL.Tree.Task,
     inputs: WDLBindings,
-    inputs_search_path: list[str],
+    inputs_search_path: List[str],
```

Comment:
@mr-c also changed to using the new generic support in the base `list`, `dict`, etc. instead of needing to import the versions from `typing`, so we shouldn't undo that.